pkg/serverless/invocationlifecycle/trace.go

```diff
@@ -27,6 +28,8 @@ const (
 	functionNameEnvVar = "AWS_LAMBDA_FUNCTION_NAME"
 )
 
+var /* const */ runtimeRegex = regexp.MustCompile(`^(java|dotnet)\d*$`)
```

Comment:
does this have to do with java and dotnet being the only universal instrumentated runtimes? What about when js and python become universal instrumented?

Reply:
i.e. what if in the future we deprecate dd-lambda-js,  I feel like we might lose the language tag. I'm just not convinced at the moment we should lock the regex down to java and dotnet but you might be able to change my mind

Reply:
I read `Since we are only adding java and dotnet language tags. And the runtime information is available already. We can safely deduce the language tag using the regex ^(java|dotnet)\d*$
`

but wouldn't the code work by just using

`executionSpan.Meta["language"] = endDetails.Runtime`

Reply:
There are a few misunderstanding here.
1. This is not a result of universal instrumentation, although people refer to dotnet and java approach as "the universal instrumentation" because... (see 2)
2. Universal instrumentation doesn't mean deprecating dd-trace-x. In fact, both dotnet and java also use dd-trace too if you think about it. 
3. "I'm just not convinced at the moment we should lock the regex down to java and dotnet"  We are not "locking" anything, this can be certainly updated to include more language needed. But I don't think it's safer to include more without any solid reason. I think it's safer to limit the scope.
4. I don't see any change will apply to the generation of aws.lambda span in javascript, python, ruby or go. Re-tagging here not only duplicates the work but also introduces uncertainty and complexity

To look at this from a high-level perspective, the task is to add the language tag to the `aws.lambda` span and because only dotnet and java's `aws.lambda` spans are generated here, we should only worry about those 2 languages's runtimes.

Reply:
> I read `Since we are only adding java and dotnet language tags. And the runtime information is available already. We can safely deduce the language tag using the regex ^(java|dotnet)\d*$ `
> 
> but wouldn't the code work by just using
> 
> `executionSpan.Meta["language"] = endDetails.Runtime`

No, see this link for explanation: https://regex101.com/r/FVjraB/1

Reply:
This will break for go and ruby. NodeJS and Python are not going to be affected by this, since we don't use this generated span for it here. If user was just using the extension, instead of the layer, they will not need this, since they don't expect runtime metrics, since they don't have the layer/tracer running there.

Fixed in: https://github.com/DataDog/datadog-agent/pull/18645